### PR TITLE
keym: init at unstable-2022-07-10

### PR DESCRIPTION
**Pull request by Compile-Time**
## Description of changes

This pull request add a simple derivation for the tool "keym": https://github.com/cwkx/keym

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by eclairevoyant — COMMENTED** [review]
Is this software maintained, or is there a better fork we can use? It seems to not have any activity for ~1.5 years

**Comment by Compile-Time** [comment]
> Is this software maintained, or is there a better fork we can use? It seems to not have any activity for ~1.5 years

I'm not aware of any alternatives. However, I also have not looked for any alternatives since I discovered this software. The closest thing that comes to my mind is [warpd](https://github.com/rvaiya/warpd) but that tool functions differently.

I think the lack of activity simply comes from the fact that the tool is complete from a functionality standpoint and it works well.

**Comment by Compile-Time** [comment]
@eclairevoyant Is the lack of activity a strong reason not to merge this?

**Comment by eclairevoyant** [comment]
I don't have merge access, just wanted to confirm, as bitrot might eventually be a higher maintinance burden. I wouldn't call it a _strong_ reason to not merge.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6960,6 +6960,8 @@ with pkgs;
 
   keymapper = callPackage ../tools/inputmethods/keymapper { };
 
+  keym = callPackage ../tools/inputmethods/keym {};
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:6963]: Please move the package to `pkgs/by-name` and remove this line

File: pkgs/tools/inputmethods/keym/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  buildInputs = [ libX11 libXtst libXi ];
+
+  makeFlags = [
+    "PREFIX=$(out)"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/cwkx/keym";
+    description = "C tool to control mouse with keyboard for X11";
+    license = licenses.mit;
```
Comment (eclairevoyant) [inline pkgs/tools/inputmethods/keym/default.nix:29]: Please add yourself to the maintainers and set `meta.mainProgram` if there's a main binary

File: pkgs/tools/inputmethods/keym/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+, libXi
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keym";
+  version = "unstable-2022-07-10";
+
+  src = fetchFromGitHub {
+    owner = "cwkx";
+    repo = pname;
```
Comment (eclairevoyant) [inline pkgs/tools/inputmethods/keym/default.nix:15]:
```suggestion
    repo = "keym";
```

pname and repo are semantically different

File: pkgs/tools/inputmethods/keym/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "67a6d39d45e17221353e06c39283b5636b46d25c";
+    sha256 = "sha256-v2eS7un2ABnpWBwuKq+0CeLX8ivtlNUjM2jRboKumOE=";
```
Comment (eclairevoyant) [inline pkgs/tools/inputmethods/keym/default.nix:17]:
```suggestion
    hash = "sha256-v2eS7un2ABnpWBwuKq+0CeLX8ivtlNUjM2jRboKumOE=";
```

File: pkgs/tools/inputmethods/keym/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, libX11
+, libXtst
+, libXi
+}:
+
+stdenv.mkDerivation rec {
```
Comment (eclairevoyant) [inline pkgs/tools/inputmethods/keym/default.nix:9]:
```suggestion
stdenv.mkDerivation {
```

**Review by eclairevoyant — DISMISSED** [review]

**Comment by Compile-Time** [comment]
@eclairevoyant I incorporated your feedback and amended them to the init commit.

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 264430` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>keym</li>
  </ul>
</details>

**Pull request merged**